File: EXAM/maza/u2/Q61-70/Q61-70/utils/MongoDBManager.js

import { MongoClient } from "mongodb";

const uri = "mongodb+srv://jmaza:jmaza@cluster0.y11rj.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0";
const client = new MongoClient(uri);

export async function connectToDatabase() {
    try {
        await client.connect();
        console.log("Connected to MongoDB successfully");
        return client.db("parks");
    } catch (error) {
        console.error("Failed to connect to MongoDB:", error);
        throw error;
    }
}

File: ws/arias/U3/WS40CreateJavaReadPython/BankInterestRate/src/main/java/utils/MongoDBManager.java
package utils;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoDatabase;
import org.bson.Document;
/**
 *
 * @author Alan Arias
 */


public class MongoDBManager {
    private static final String URI = "mongodb+srv://ajarias:ajarias@cluster0.8efem.mongodb.net/?retryWrites=true&w=majority";

    public void connect() {
        try {
            MongoClient client = MongoClients.create(URI);
            MongoDatabase database = client.getDatabase("BankDB");

            System.out.println("Successfully connected to MongoDB");

            Document user = new Document("name", "User")
                    .append("capital", 1000)
                    .append("interestRate", 5)
                    .append("time", 1);

            database.getCollection("users").insertOne(user);

        } catch (Exception e) {
            System.err.println("Error connecting to MongoDB: " + e.getMessage());
        }
    }
}